app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -340,51 +326,52 @@ class MovementsRepositoryImpl @Inject() (
       movementMRNFilter(movementReferenceNumber),
       movementLRNFilter(localReferenceNumber)
     )
-    val messageSummaryQuery = Json.obj("$ifNull" -> Json.arr("$movementSummary", "[]"))
-
     val (from, itemCount) = indices(page, count)
-
-    val aggregates = Seq(
-      Aggregates.filter(selector),
+    val filterAggregates  = Seq(Aggregates.filter(selector))
+    val aggregates = filterAggregates ++ Seq(
       Aggregates.sort(descending("updated")),
-      Aggregates.project(Projections.exclude("messages")),
-      Aggregates.facet(
-        Facet("totalCount", Aggregates.count()),
-        Facet("movementSummary", Aggregates.skip(from), Aggregates.limit(itemCount))
-      ),
-      Aggregates.project(Document("totalCount" -> Codecs.toBson(totalCountQuery()), "movementSummary" -> Codecs.toBson(messageSummaryQuery)))
+      Aggregates.skip(from),
+      Aggregates.limit(itemCount)
     )
+    for {
+      perPageMovements <- filterPerPageMovements(aggregates)
+      totalCount       <- totalCountMovements(filterAggregates)
+    } yield PaginationMovementSummary(TotalCount(totalCount), perPageMovements)
+
+  }
 
-    mongoRetry(Try(collection.aggregate[PaginationMovementSummary](aggregates)) match {
+  private def filterPerPageMovements(aggregates: Seq[Bson]): EitherT[Future, MongoError, Vector[MovementWithoutMessages]] =
+    mongoRetry(Try(collection.aggregate[MovementWithoutMessages](aggregates)) match {
       case Success(obs) =>
         obs
           .toFuture()
           .map {
-            response => Right(response.head)
+            response => Right(response.toVector)
           }
 
       case Failure(NonFatal(ex)) =>
         Future.successful(Left(UnexpectedError(Some(ex))))
     })
 
-  }
+  private def totalCountMovements(totalDocument: Seq[Bson]): EitherT[Future, MongoError, Long] =
+    mongoRetry(Try(collection.aggregate[MovementWithoutMessages](totalDocument)) match {
+      case Success(obs) =>
+        obs.toFuture().map {
+          response => Right(response.size.toLong)
+        }
+      case Failure(NonFatal(ex)) =>
+        Future.successful(Left(UnexpectedError(Some(ex))))
+    })
 
-  private def totalCountQuery() =
-    Json.obj(
-      "$ifNull" -> Json.arr(
-        Json.obj(
-          "$let" -> Json.obj(
-            "vars" -> Json.obj(
-              "countValue" -> Json.obj(
-                "$arrayElemAt" -> Json.arr("$totalCount", 0)
-              )
-            ),
-            "in" -> "$$countValue.count"
-          )
-        ),
-        0
-      )
-    )
+  private def totalCountMessages(totalDocument: Seq[Bson]): EitherT[Future, MongoError, Long] =
```

Comment:
This is just a duplicate of the above function except for the type that is returned, but by using the `$count` aggregation, that distinction will disappear too.

I suggest removing this method and renaming the other one to something like "countItems".